src/scheduler/Scheduler.cpp

```diff
@@ -224,228 +223,339 @@ faabric::util::SchedulingDecision Scheduler::callFunctions(
         throw std::runtime_error("Message with no master host");
     }
 
-    // Set up scheduling decision
```

Comment:
The diff of the `callFunctions` methods isn't particularly easy to parse, full version [here](https://github.com/faasm/faabric/blob/dist-lock-fixes/src/scheduler/Scheduler.cpp).